File: Typescript/strings/customSortString.ts
var solution = function (order: string, s: string): string {
    let res = ""
    let temp: [string, number][] = [] 
    for (let char of s){
        if (order.includes(char)){
            let idx = order.indexOf(char)
            temp.push([char, idx])
        }
    }

    temp.sort((a,b) => a[1] - b[1])
    for (let i = 0; i< temp.length; i++){
        res += temp[i][0]
    }

    console.log(res)    
    let idx = 0
    let length = s.length

    while (idx < length) {
        let val = s[idx]
        if (!order.includes(val)){
            res += val
        }
        idx ++
    }

    return res
};

File: Python/Medium/strings/customSortString.py
def solution(order, s):
    temp = []
    res = ""
    for char in s:
        if char in order:
            idx = order.index(char)
            temp.append((char, idx))

    temp = list(sorted(temp , key = lambda x: x[1]))
    for val in temp:
        res += val[0]

    idx = 0
    length = len(s)

    while idx < length:
        val = s[idx]
        if val not in order:
            res += val
    
    return val

